src/server/routes/collection.js

```diff
@@ -56,16 +107,72 @@ router.param(
 	middleware.makeCollectionLoader()
 );
 
-router.post('/:collectionId/delete/handler', auth.isAuthenticatedForHandler, auth.isCollectionOwner, async (req, res) => {
+router.get('/:collectionId', auth.isAuthenticatedForCollectionView, async (req, res, next) => {
 	try {
-		const {UserCollection} = req.app.locals.orm;
-		const {collectionId} = req.params;
-		await new UserCollection({id: collectionId}).destroy();
-		return res.status(200).send({});
+		const {collection} = res.locals;
+		const {orm} = req.app.locals;
+		const size = req.query.size ? parseInt(req.query.size, 10) : 20;
+		const from = req.query.from ? parseInt(req.query.from, 10) : 0;
+		const relations = getEntityRelations(collection.entityType);
+
+		// fetch 1 more bbid to check next enabled for pagination
+		const bbids = await getCollectionItemBBIDs(collection.id, from, size + 1, orm);
+		// get next enabled for pagination
+		const {newResultsArray, nextEnabled} = utils.getNextEnabledAndResultsArray(bbids, size);
+		// load entities from bbids
+		const entitiesPromise = newResultsArray.map(bbid => orm.func.entity.getEntity(orm, collection.entityType, bbid, relations));
+		const entities = await Promise.all(entitiesPromise);
+		const isOwner = req.user && parseInt(collection.ownerId, 10) === parseInt(req.user.id, 10);
```

Comment:
I have made this change, but I don't think this was needed. We're checking `req.user` first and because of `&&` if the first query is false the second is not calculated.

Reply:
True.